docs/6-secret-management-in-saap/6-secret-management.md

```diff
@@ -1,18 +1,337 @@
 
 ## Secrets creation workflow
-_TODO_
+
+```mermaid 
+sequenceDiagram
+    autonumber
+    actor User
+    participant Vault
+    participant SecretStore
+    participant ExternalSecret
+    participant ESO as External Secret Operator 
+    participant k8s Secret
+    participant app as Application
+    User->>Vault: Adds key/value pair secret
+    User->>ExternalSecret: Adds ExternalSecret CR
+    ESO->>+ExternalSecret: watches for CR creation
+    ExternalSecret-->>-ESO: CR created, ESO to reconcile
+    ESO-->SecretStore: ESO uses defined SecretStore & instanciates Vault request using ServiceAccount for authentication
+    ESO->>+Vault: Requests to fetch secret data from path
+    Vault->>-ESO: Returns secret data
+    ESO->>k8s Secret: Creates a k8s Secret
+    k8s Secret ->> app: Secret is used by Application
+```
+
+   ### Workflow 
+
+   1. User adds a Key/Value pair secret data to a path in Vault.
+   
+      - In the path of your tenant, Click `Create Secret`, add path of secret, and add key-value pairs.
+
+  ![create-secret](./images/create-secret.png)
+
+   2. Add ExternalSecret CR
+
+      - Login to the `OpenShift console`.
+      - Select the `+` sign in the top right corner of the console
+
+      ![the-plus-sign](./images/the-plus-sign.png)
+      - Paste the following YAML and replace the `<TENANT_NAME>` and click `Create`
+
+   ```
+    apiVersion: external-secrets.io/v1alpha1
+    kind: ExternalSecret
+    metadata:
+      name: review-ui-secret
+      namespace: <TENANT_NAME>-build
+    spec:
+      secretStoreRef:
+        kind: SecretStore
+        name: tenant-vault-secret-store
+      refreshInterval: "1m"
+      target:
+        name: review-ui
+        creationPolicy: 'Owner'
+        deletionPolicy: Retain
+        template:
+          type: kubernetes.io/basic-auth
+          metadata:
+            annotations:
+              tekton.dev/git-0: 'https://gitlab.apps.devtest.vxdqgl7u.kubeapp.cloud'
+      dataFrom:
+        - key: nordmart-review-ui-page-title
+
+   ```
+   3. External Secrets Operator (ESO) watches for ExternalSecret CR creation.
+
+   4. When a new CR is created, External Secrets Operator (ESO) reconciles it from the provided spec.
+
+   5. External Secrets Operator (ESO) uses the defined SecretStore from ExternalSecret CR and instantiates Vault request using ServiceAccount created, and bound to Vault role, to authenticate to Vault. 
+
+      ```
+      secretStoreRef:
+         kind: SecretStore
+         name: tenant-vault-secret-store     
+      ```
+
+   6. External Secrets Operator (ESO) requests to fetch secret data from the path specified in ExternalSecret CR.
+
+      ```
+         dataFrom:
+            - key: gitlab-pat
+      ```
+
+   7. Vault returns the requested secret data to External Secrets Operator (ESO).
+
+   8. External Secrets Operator (ESO) creates a Kubernetes Secret. 
+
+   9. Kubernetes Secret is consumed by Application.
```

Comment:
add step for 
Adding the secret in application values.yaml by specifying `deployment.env.PAGE_TITLE` with valueFrom field referencing this secret

add step for `helm template deploy/ -n <TENANT>-dev | oc apply -f -`

add step for showing the updated ui